<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>819</width>
    <height>793</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Observer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_8">
    <item>
     <widget class="QVTKOpenGLWidget" name="Display_Window">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>801</width>
        <height>431</height>
       </size>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="sizeConstraint">
        <enum>QLayout::SetFixedSize</enum>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QGroupBox" name="groupBox">
               <property name="title">
                <string>Position</string>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_9">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_2">
                  <item>
                   <widget class="QPushButton" name="Reset_Camera">
                    <property name="minimumSize">
                     <size>
                      <width>76</width>
                      <height>23</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Camera Reset</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="maximumSize">
                     <size>
                      <width>78</width>
                      <height>47</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Vertical Shift</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="Vertical_Shift">
                    <property name="minimum">
                     <number>-999</number>
                    </property>
                    <property name="maximum">
                     <number>999</number>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="minimumSize">
                     <size>
                      <width>78</width>
                      <height>47</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Horizontal Shift</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="Horizontal_Shift">
                    <property name="minimum">
                     <number>-999</number>
                    </property>
                    <property name="maximum">
                     <number>999</number>
                    </property>
                    <property name="stepType">
                     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <spacer name="horizontalSpacer">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <widget class="QGroupBox" name="groupBox_2">
               <property name="title">
                <string>Object View</string>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_10">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <item>
                   <widget class="QPushButton" name="Change_Object_Color">
                    <property name="minimumSize">
                     <size>
                      <width>75</width>
                      <height>23</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Model Colour</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="Change_Back_Ground_Color">
                    <property name="text">
                     <string>Background Colour</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="Apply_Filters">
                    <property name="minimumSize">
                     <size>
                      <width>75</width>
                      <height>23</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Filters</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <spacer name="horizontalSpacer_3">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string>Light</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QPushButton" name="Add_Light">
                  <property name="minimumSize">
                   <size>
                    <width>75</width>
                    <height>23</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Add Light</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="Delete_Light">
                  <property name="minimumSize">
                   <size>
                    <width>75</width>
                    <height>23</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Delete Light</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="Edit_Light">
                  <property name="minimumSize">
                   <size>
                    <width>75</width>
                    <height>23</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Edit Light</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QComboBox" name="Select_Light">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <property name="sizeConstraint">
          <enum>QLayout::SetFixedSize</enum>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>Rotation</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_12">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLCDNumber" name="lcdnumber">
                <property name="maximumSize">
                 <size>
                  <width>78</width>
                  <height>23</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>X-Plane</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDial" name="X_Camera_Pos">
                <property name="maximumSize">
                 <size>
                  <width>61</width>
                  <height>51</height>
                 </size>
                </property>
                <property name="minimum">
                 <number>-360</number>
                </property>
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="wrapping">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Y-Plane</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDial" name="Y_Camera_Pos">
                <property name="maximumSize">
                 <size>
                  <width>61</width>
                  <height>51</height>
                 </size>
                </property>
                <property name="minimum">
                 <number>-360</number>
                </property>
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="wrapping">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Z-Plane</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDial" name="Z_Camera_Pos">
                <property name="maximumSize">
                 <size>
                  <width>61</width>
                  <height>51</height>
                 </size>
                </property>
                <property name="minimum">
                 <number>-360</number>
                </property>
                <property name="wrapping">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>819</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionLoad_Lights"/>
    <addaction name="actionSave_Lights"/>
   </widget>
   <widget class="QMenu" name="menuViews">
    <property name="title">
     <string>Views</string>
    </property>
    <widget class="QMenu" name="menuCube">
     <property name="title">
      <string>Coordinate axes</string>
     </property>
     <addaction name="actioncube"/>
     <addaction name="actionhelicopter"/>
     <addaction name="actionairbus"/>
     <addaction name="actionplane"/>
     <addaction name="actionsphere"/>
     <addaction name="actionThunderbolt"/>
    </widget>
    <addaction name="menuCube"/>
    <addaction name="separator"/>
    <addaction name="actionRuler"/>
    <addaction name="actionRemove_Ruler"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuViews"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionLoad_Lights"/>
   <addaction name="actionSave_Lights"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>../resources/images/Open.png</normaloff>
     <normalon>../images/Open.png</normalon>../resources/images/Open.png</iconset>
   </property>
   <property name="text">
    <string>Open Model</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>../resources/images/Save_Icon.png</normaloff>
     <normalon>../images/Save_Icon.png</normalon>../resources/images/Save_Icon.png</iconset>
   </property>
   <property name="text">
    <string>Save Model</string>
   </property>
  </action>
  <action name="actionSave_Lights">
   <property name="icon">
    <iconset>
     <normalon>../images/spotlight.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Save Lights</string>
   </property>
  </action>
  <action name="actionLoad_Lights">
   <property name="icon">
    <iconset>
     <normalon>../images/business-and-finance.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Load Lights</string>
   </property>
  </action>
  <action name="actioncube">
   <property name="text">
    <string>cube</string>
   </property>
  </action>
  <action name="actionhelicopter">
   <property name="text">
    <string>helicopter</string>
   </property>
  </action>
  <action name="actionairbus">
   <property name="text">
    <string>airbus</string>
   </property>
  </action>
  <action name="actionplane">
   <property name="text">
    <string>bawa plane</string>
   </property>
  </action>
  <action name="actionsphere">
   <property name="text">
    <string>sphere</string>
   </property>
  </action>
  <action name="actionThunderbolt">
   <property name="text">
    <string>Thunderbolt</string>
   </property>
  </action>
  <action name="actionRuler">
   <property name="text">
    <string>Ruler</string>
   </property>
  </action>
  <action name="actionRemove_Ruler">
   <property name="text">
    <string>Remove Ruler</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKOpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>qvtkopenglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>X_Camera_Pos</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdnumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>709</y>
    </hint>
    <hint type="destinationlabel">
     <x>62</x>
     <y>722</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Z_Camera_Pos</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdnumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>703</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>714</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Y_Camera_Pos</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdnumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>719</y>
    </hint>
    <hint type="destinationlabel">
     <x>49</x>
     <y>711</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
